### PR TITLE
feat: Add KeysInValidator for validating fields containing specific keys

### PR DESCRIPTION
**Pull request by bangadam**
## References
Issue(s): #201 
Discussion: None

## Description
This validator ensures that all keys in the object under validation are part of the specified parameter slice. The main benefits of this addition are:
- It provides a simple and efficient way to validate that an object contains only the allowed keys, which enhances data integrity and consistency.
- It reduces the need for custom validation logic in individual projects, promoting code reusability and maintainability.

### Possible drawbacks
- The KeysIn validator assumes that the object being validated is a map[string]interface{}. If the context value is not of this type, the validation will fail

### CODE REVIEW

File: validation/keysin_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package validation
+
+import "testing"
+
+func TestKeysInValidator_Validate(t *testing.T) {
```
Comment (System-Glitch) [inline validation/keysin_test.go:5]: Please add a test for the validator constructor (checking it's `Name()`, `IsType()`, `IsTypeDependent()` and `MessagePlaceholders()`.

File: validation/keysin_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v := &KeysInValidator{
+				BaseValidator: tt.fields.BaseValidator,
+				Keys:          tt.fields.Keys,
+			}
+			if got := v.Validate(tt.args.ctx); got != tt.want {
+				t.Errorf("KeysInValidator.Validate() = %v, want %v", got, tt.want)
```
Comment (System-Glitch) [inline validation/keysin_test.go:68]: You can use `testify` for assertions.

File: validation/keysin.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	obj, ok := ctx.Value.(map[string]interface{})
+	if !ok {
+		return false
+	}
+
+	for _, key := range v.Keys {
+		if _, ok := obj[key]; !ok {
+			return false
+		}
+	}
```
Comment (System-Glitch) [inline validation/keysin.go:22]: This validator should check that the object doesn't contain a key that is not allowed. Here, you are checking that the object contains all the keys, which is different.

**Review by System-Glitch — CHANGES_REQUESTED** [review]
You forgot to add the default language lines in `lang/default.go`.

**Comment by System-Glitch** [comment]
> The KeysIn validator assumes that the object being validated is a map[string]interface{}. If the context value is not of this type, the validation will fail

It is correct and this is how it should be done. 👍🏻

**Comment by System-Glitch** [comment]
Implementation looks correct to me. The third test case should expect `true` instead of `false`.

**Comment by bangadam** [comment]
> Implementation looks correct to me. The third test case should expect `true` instead of `false`.

updated, please check

File: validation/string.go [hunk 4]
```diff
@@ -124,3 +124,46 @@ func (v *DoesntStartWithValidator) MessagePlaceholders(_ *Context) []string {
 func DoesntStartWith(prefix ...string) *DoesntStartWithValidator {
 	return &DoesntStartWithValidator{Prefix: prefix}
 }
+
+//------------------------------
+
+// KeysInValidator validates the field under validation must only contain the given keys.
```
Comment (System-Glitch) [inline validation/string.go:130]:
```suggestion
// KeysInValidator the field under validation must be an object and all its keys must
// be equal to one of the given values.
```

File: validation/string.go [hunk 5]
```diff
@@ -124,3 +124,46 @@ func (v *DoesntStartWithValidator) MessagePlaceholders(_ *Context) []string {
...
+func (v *KeysInValidator) Name() string { return "keys_in" }
+
+// MessagePlaceholders returns the ":values" placeholder.
+func (v *KeysInValidator) MessagePlaceholders(_ *Context) []string {
+	return []string{
+		":values", strings.Join(v.Keys, ", "),
+	}
+}
+
+// KeysIn creates a new KeysInValidator.
```
Comment (System-Glitch) [inline validation/string.go:166]:
```suggestion
// KeysIn the field under validation must be an object and all its keys must
// be equal to one of the given values.
```

File: validation/string_test.go [hunk 6]
```diff
@@ -168,3 +168,34 @@ func TestDoesntStartWithValidator(t *testing.T) {
...
+		assert.False(t, v.IsType())
+		assert.False(t, v.IsTypeDependent())
+		assert.Equal(t, []string{":values", "a, b, c"}, v.MessagePlaceholders(&Context{}))
+	})
+
+	cases := []struct {
+		value []string
+		input any
+		want  bool
+	}{
```
Comment (System-Glitch) [inline validation/string_test.go:186]:
```suggestion
	cases := []struct {
		input any
		value []string
		want  bool
	}{
```

File: validation/string_test.go [hunk 7]
```diff
@@ -168,3 +168,34 @@ func TestDoesntStartWithValidator(t *testing.T) {
...
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3}, want: true},
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3, "d": 4}, want: false},
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2}, want: true},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(fmt.Sprintf("Validate_%v_%t", tc.value, tc.want), func(t *testing.T) {
```
Comment (System-Glitch) [inline validation/string_test.go:194]:
```suggestion
		t.Run(fmt.Sprintf("Validate_%v_%t", tc.input, tc.want), func(t *testing.T) {
```

File: validation/string_test.go [hunk 8]
```diff
@@ -168,3 +168,34 @@ func TestDoesntStartWithValidator(t *testing.T) {
 		})
 	}
 }
+
+func TestKeysInValidator(t *testing.T) {
+	t.Run("Constructor", func(t *testing.T) {
+		v := KeysIn([]string{"a", "b", "c"}...)
```
Comment (System-Glitch) [inline validation/string_test.go:174]:
```suggestion
		v := KeysIn("a", "b", "c")
```

File: validation/string_test.go [hunk 9]
```diff
@@ -168,3 +168,34 @@ func TestDoesntStartWithValidator(t *testing.T) {
...
+	})
+
+	cases := []struct {
+		value []string
+		input any
+		want  bool
+	}{
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3}, want: true},
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3, "d": 4}, want: false},
+		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2}, want: true},
```
Comment (System-Glitch) [inline validation/string_test.go:189]:
```suggestion
		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3}, want: true},
		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2, "c": 3, "d": 4}, want: false},
		{value: []string{"a", "b", "c"}, input: map[string]any{"a": 1, "b": 2}, want: true},
		{value: []string{"a", "b", "c"}, input: map[string]any{}, want: true},
		{value: []string{"a", "b", "c"}, input: "", want: false},
		{value: []string{"a", "b", "c"}, input: 'a', want: false},
		{value: []string{"a", "b", "c"}, input: 2, want: false},
		{value: []string{"a", "b", "c"}, input: 2.5, want: false},
		{value: []string{"a", "b", "c"}, input: true, want: false},
		{value: []string{"a", "b", "c"}, input: nil, want: false},
		{value: []string{"a", "b", "c"}, input: (map[string]any)(nil), want: true},
```

**Review by System-Glitch — APPROVED** [review]

**Comment by System-Glitch** [comment]
Thank you very much! 🎉

**Pull request merged**